pkg/graph/resolvers/key_contact.go

```diff
@@ -0,0 +1,320 @@
+package resolvers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/zap"
+
+	"github.com/cms-enterprise/mint-app/pkg/authentication"
+	"github.com/cms-enterprise/mint-app/pkg/email"
+	"github.com/cms-enterprise/mint-app/pkg/models"
+	"github.com/cms-enterprise/mint-app/pkg/shared/oddmail"
+	"github.com/cms-enterprise/mint-app/pkg/sqlutils"
+	"github.com/cms-enterprise/mint-app/pkg/storage"
+	"github.com/cms-enterprise/mint-app/pkg/storage/loaders"
+	"github.com/cms-enterprise/mint-app/pkg/userhelpers"
+)
+
+// CreateKeyContactUser creates a new user contact for a subject matter expert.
+// It looks up the user account by username and inserts a new contact record associated with that user.
+func CreateKeyContactUser(ctx context.Context, logger *zap.Logger, principal authentication.Principal, store *storage.Store,
+	emailService oddmail.EmailService, emailTemplateService email.TemplateService, addressBook email.AddressBook,
+	userName string,
+	subjectArea string,
+	subjectCategoryID uuid.UUID,
+	getAccountInformation userhelpers.GetAccountInfoFunc,
+) (*models.KeyContact, error) {
+	principalAccount := principal.Account()
+	if principalAccount == nil {
+		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
+	}
+
+	userAccount, err := userhelpers.GetOrCreateUserAccount(ctx, store, store, userName, false, false, getAccountInformation)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get user account by username %s: %w", userName, err)
+	}
+
+	userContact := models.NewKeyContact(
+		principalAccount.ID,
+		nil,
+		nil,
+		&userAccount.ID,
+		subjectArea,
+		subjectCategoryID,
+	)
+
+	err = BaseStructPreCreate(logger, userContact, principal, store, false)
+	if err != nil {
+		return nil, err
+	}
+
+	newContact, err := sqlutils.WithTransaction(store, func(tx *sqlx.Tx) (*models.KeyContact, error) {
```

Comment:
I don't think we need a transaction here if it is only 1 call to the DB, transactions are generally used when you are making multiple calls and if any fail you want to rollback everything